lib/game.rb

```diff
@@ -0,0 +1,93 @@
+class Game
+
+    attr_reader :score, :frame_counter
+    FRAME_MAX = 10
+    BONUS_VALUE = 10
+
+    def initialize
+        @score = 0
+        @frame_counter = 0
+        @spare_counter = 0
+        @extra_roll = 0
+        @results = []
+        @frame = []
+    end
+
+    def roll(pins)
+        fail 'This game is over' if @frame_counter == FRAME_MAX
```

Comment:
Good idea to let the player know when the game is over rather than just stopping the game.